ClosedXML/Excel/PivotTables/XLPivotTable.cs

```diff
@@ -80,6 +81,122 @@ public IEnumerable<IXLPivotField> ImplementedFields
+
+            int i = 0;
+            while (!XLHelper.ValidateName("pivot table", pivotTableName, "", targetSheet.PivotTables.Select(pvt => pvt.Name), out String message))
+            {
+                i++;
+                pivotTableName = this.Name + i.ToInvariantString();
+            }
+
+            XLPivotTable newPivotTable;
+            if (this.SourceType == XLPivotTableSourceType.Table)
+                newPivotTable = targetSheet.PivotTables.Add(pivotTableName, targetCell, this.SourceTable) as XLPivotTable;
+            else if (this.SourceType == XLPivotTableSourceType.Range)
+                newPivotTable = targetSheet.PivotTables.Add(pivotTableName, targetCell, this.SourceRange) as XLPivotTable;
+            else
+                throw new NotImplementedException();
+
+            newPivotTable.RelId = null;
+
+            newPivotTable.Theme = Theme;
+
+            IXLPivotField copyPivotField(IXLPivotField originalPivotField, IXLPivotField newPivotField)
+            {
+                newPivotField
+                    .SetSort(originalPivotField.SortType)
+                    .SetSubtotalCaption(originalPivotField.SubtotalCaption)
+                    .SetIncludeNewItemsInFilter(originalPivotField.IncludeNewItemsInFilter)
+                    .SetRepeatItemLabels(originalPivotField.RepeatItemLabels)
+                    .SetInsertBlankLines(originalPivotField.InsertBlankLines)
+                    .SetShowBlankItems(originalPivotField.ShowBlankItems)
+                    .SetInsertPageBreaks(originalPivotField.InsertPageBreaks)
+                    .SetCollapsed(originalPivotField.Collapsed);
+
+                if (originalPivotField.SubtotalsAtTop.HasValue)
+                    newPivotField.SetSubtotalsAtTop(originalPivotField.SubtotalsAtTop.Value);
+
+                foreach (var v in originalPivotField.SelectedValues)
+                    newPivotField.AddSelectedValue(v);
+
+                return newPivotField;
+            }
+
+            foreach (var rf in ReportFilters)
+                copyPivotField(rf, newPivotTable.ReportFilters.Add(rf.SourceName, rf.CustomName));
+
+            foreach (var cl in ColumnLabels)
+                copyPivotField(cl, newPivotTable.ColumnLabels.Add(cl.SourceName, cl.CustomName));
+
+            foreach (var rl in RowLabels)
+                copyPivotField(rl, newPivotTable.RowLabels.Add(rl.SourceName, rl.CustomName));
+
+            foreach (var v in Values)
+            {
+                var pivotValue = newPivotTable.Values.Add(v.SourceName, v.CustomName)
+                    .SetSummaryFormula(v.SummaryFormula)
+                    .SetCalculation(v.Calculation)
+                    .SetCalculationItem(v.CalculationItem)
+                    .SetBaseField(v.BaseField)
+                    .SetBaseItem(v.BaseItem);
+
+                pivotValue.NumberFormat.NumberFormatId = v.NumberFormat.NumberFormatId;
+                pivotValue.NumberFormat.Format = v.NumberFormat.Format;
```

Comment:
Can't we define `.SetNumberFormat(numberFormatId, format)`?

Reply:
`NumberFormatId` and `Format` are actually not meant to use in conjuction. The format ID is supposed to be in the range of the predefined number format IDs (something like 1 to 164?). The explicit format string is to be used for a custom format that is not in the predefined formats. So typically, either `NumberFormatId == 0` or `Format == ""` . Therefore I don't really see the use of `.SetNumberFormat(numberFormatId, format)`, at least not in the public API.